mmv1/products/iamworkforcepool/terraform.yaml

```diff
@@ -42,6 +42,56 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       workforcePoolId: !ruby/object:Overrides::Terraform::PropertyOverride
         validation: !ruby/object:Provider::Terraform::Validation
           function: 'validateWorkforcePoolId'
+  WorkforcePoolProvider: !ruby/object:Overrides::Terraform::ResourceOverride
+    autogen_async: true
+    import_format: ["locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers/{{provider_id}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "iam_workforce_pool_provider_saml_basic"
+        min_version: beta
+        primary_resource_id: "example"
+        vars:
+          workforce_pool_id: "example-pool"
+          provider_id: "example-prvdr"
+        test_env_vars:
+          org_id: :ORG_ID
+      - !ruby/object:Provider::Terraform::Examples
+        name: "iam_workforce_pool_provider_saml_full"
+        min_version: beta
+        primary_resource_id: "example"
+        vars:
+          workforce_pool_id: "example-pool"
+          provider_id: "example-prvdr"
+        test_env_vars:
+          org_id: :ORG_ID
+      - !ruby/object:Provider::Terraform::Examples
+        name: "iam_workforce_pool_provider_oidc_basic"
+        min_version: beta
+        primary_resource_id: "example"
+        vars:
+          workforce_pool_id: "example-pool"
+          provider_id: "example-prvdr"
+        test_env_vars:
+          org_id: :ORG_ID
+      - !ruby/object:Provider::Terraform::Examples
+        name: "iam_workforce_pool_provider_oidc_full"
+        min_version: beta
+        primary_resource_id: "example"
+        vars:
+          workforce_pool_id: "example-pool"
+          provider_id: "example-prvdr"
+        test_env_vars:
+          org_id: :ORG_ID
+    custom_code: !ruby/object:Provider::Terraform::CustomCode
+      constants: templates/terraform/constants/iam_workforce_pool_provider.go.erb
+      decoder: templates/terraform/decoders/treat_deleted_state_as_gone.go.erb
+    properties:
+      providerId: !ruby/object:Overrides::Terraform::PropertyOverride
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validateWorkforcePoolProviderId'
+      saml: !ruby/object:Overrides::Terraform::PropertyOverride
+        update_mask_fields:
+          - "saml.idp_metadata_xml"
```

Comment:
do you need to set update_mask_fields for oidc as well?

Reply:
It looks like the update tests for oidc are able to work fine without needing update_mask_fields.

Do you need it for `saml`? I would be a little surprised if oidc and saml have different update behavior.

Reply:
fixed. There was a bug in our backend, but it has been fixed, so I removed these lines.